tests/backup/backup_restore_basic_test.go

```diff
@@ -3842,3 +3843,333 @@ var _ = Describe("{KubeAndPxNamespacesSkipOnAllNSBackup}", func() {
+		dirListMountMap               map[string][]string
+		formattedFileString           string
+		fileList                      []string
+		dirList                       []string
+		numOfFiles                    int
+		depthOfdirs                   int
+		levelOfdirs                   int
+		preRuleName                   string
+		postRuleName                  string
+		preRuleUid                    string
+		postRuleUid                   string
+	)
+	JustBeforeEach(func() {
+		backupNames = make([]string, 0)
+		restoreNames = make([]string, 0)
+		bkpNamespaces = make([]string, 0)
+		backupLocationMap = make(map[string]string)
+		labelSelectors = make(map[string]string)
+		AppContextsMapping = make(map[string]*scheduler.Context)
+		storageClassExcludeFileDirMap = make(map[string][]string)
+		mountPathExcludeFileDirMap = make(map[string][]string)
+		existingFileDirMountPathMap = make(map[string][]string)
+		podMountPathMap = make(map[string][]string)
+		fileListMountMap = make(map[string][]string)
+		dirListMountMap = make(map[string][]string)
+		numOfFiles = 5
+		depthOfdirs = 5
+		levelOfdirs = 0
+
+		numDeployments = 1
+		providers = getProviders()
+
+		StartPxBackupTorpedoTest("ExcludeDirectoryFileBackup", "Excludes mentioned directories or files from backed-up apps and restores them", nil, 0, Ak, Q4FY24)
+
+		log.InfoD(fmt.Sprintf("App list %v", Inst().AppList))
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		log.InfoD("Starting to deploy applications")
+		for i := 0; i < numDeployments; i++ {
+			log.InfoD(fmt.Sprintf("Iteration %v of deploying applications", i))
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts := ScheduleApplications(taskName)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace = GetAppNamespace(ctx, taskName)
+				scheduledAppContexts = append(scheduledAppContexts, ctx)
+				AppContextsMapping[namespace] = ctx
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+
+	})
+	It("Excludes directories or files From a Backup", func() {
+
+		Step("Validating deployed applications", func() {
+			log.InfoD("Validating deployed applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+
+		Step("Getting mountpath for deployed application ", func() {
+			log.InfoD("Getting mountpath for deployed application ")
+			for _, namespace := range bkpNamespaces {
+				pods, err := core.Instance().GetPods(namespace, nil)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("getting pods from namespace [%s] ", namespace))
+				for _, pod := range pods.Items {
+					containerMountPathList := make([]string, 0)
+					containerPaths := schedops.GetContainerPVCMountMap(pod)
+					for containerName, paths := range containerPaths {
+						log.Infof(fmt.Sprintf("containerName:%s,paths: %s ", containerName, paths))
+						containerMountPathList = append(containerMountPathList, paths...)
+					}
+					if containerMountPathList != nil {
```

Comment:
Shouldn't you check the length of `containerMountPathList` here instead of nil check?